### ISSUE TITLE
QGIS freezes when creating new shapefiles

### ISSUE THREAD

**Issue by PedroVenancio** [qgis/QGIS#32069]
**How to Reproduce**
1. Layer -> Create Layer -> New Shapefile Layer.
2. Choose folder and name to save the new shapefile, and let everything else as default (point, field "id", ...)
3. Create a new point geometry anywhere. Fill "id" field, for instance, with 1 or anything else.
4. Open attributes table.
5. New field - Any name and type. Ok.
6. New field - Any name and type. Ok 

QGIS freezes.

If I do not create the new point at step 3, the freeze does not happen and all goes ok.

**QGIS and OS versions**

Tested in Windows, on 3.4, 3.8 and master.

**Comment by gioman** [comment]
seen also on Linux.

**Comment by nyalldawson** [comment]
I can't reproduce -- could someone share a screencast?

**Comment by PedroVenancio** [comment]
Hi @nyalldawson 

I was looking further, and the freeze only happens when, in the step 3, I save the point in Save Layers Edit button, in Digitizing Toolbar. Please see the screencast:

![qgis_freeze](https://user-images.githubusercontent.com/1331581/65949462-27e0ff80-e434-11e9-8487-6c285ef9c882.gif)


```
Crash ID: 8637079635e2ae56dbeada11e1d33378814377d5

Stack Trace

memcpy_repmovs memcpy.asm:115
QSortFilterProxyModel::qt_metacast :
QSortFilterProxyModel::removeRows :
QSortFilterProxyModel::sortRole :
QSortFilterProxyModel::`default constructor closure' :
QMetaObject::activate :
QAbstractItemModel::beginRemoveRows :
QgsAttributeTableModel::removeRows qgsattributetablemodel.cpp:171
QgsAttributeTableModel::loadLayer qgsattributetablemodel.cpp:448
QtPrivate::FunctorCall,QtPrivate::List,void,void (__cdecl QgsAttributeTableModel::*)(void) __ptr64>::call qobjectdefs_impl.h:134
QtPrivate::FunctionPointer::call,void> qobjectdefs_impl.h:168
QtPrivate::QSlotObject,void>::impl qobjectdefs_impl.h:397
QMetaObject::activate :
QgsVectorLayerCache::invalidated moc_qgsvectorlayercache.cpp:271
QgsVectorLayerCache::invalidate qgsvectorlayercache.cpp:328
QtPrivate::FunctorCall,QtPrivate::List,void,void (__cdecl QgsVectorLayerCache::*)(void) __ptr64>::call qobjectdefs_impl.h:134
QtPrivate::FunctionPointer::call,void> qobjectdefs_impl.h:168
QtPrivate::QSlotObject,void>::impl qobjectdefs_impl.h:397
QMetaObject::activate :
QgsVectorLayer::updatedFields moc_qgsvectorlayer.cpp:949
QgsVectorLayer::updateFields qgsvectorlayer.cpp:3529
QgsVectorLayerEditBuffer::updateLayerFields qgsvectorlayereditbuffer.cpp:826
QgsVectorLayerUndoCommandAddAttribute::redo qgsvectorlayerundocommand.cpp:317
QUndoStack::push :
QgsVectorLayerEditBuffer::addAttribute qgsvectorlayereditbuffer.cpp:274
QgsVectorLayer::addAttribute qgsvectorlayer.cpp:2779
QgsAttributeTableDialog::mActionAddAttribute_triggered qgsattributetabledialog.cpp:963
QtPrivate::FunctorCall,QtPrivate::List,void,void (__cdecl QgsAttributeTableDialog::*)(void) __ptr64>::call qobjectdefs_impl.h:134
QtPrivate::FunctionPointer::call,void> qobjectdefs_impl.h:168
QtPrivate::QSlotObject,void>::impl qobjectdefs_impl.h:397
QMetaObject::activate :
QAction::activate :
QAbstractButton::click :
QAbstractButton::mouseReleaseEvent :
QToolButton::mouseReleaseEvent :
QWidget::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:403
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:403
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main main.cpp:1567
WinMain mainwin.cpp:170
__scrt_common_main_seh exe_common.inl:253
BaseThreadInitThunk :
RtlUserThreadStart :

QGIS Info
QGIS Version: 3.8.3-Zanzibar
QGIS code revision: 0b21735e28
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 2.4.1
Running against GDAL: 2.4.1

System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.18362
```

Saving the geometry in Save Edits button, inside attributes table, I don't get the freeze.

Can you reproduce?

**Comment by PedroVenancio** [comment]
This is still true in QGIS 3.4.13 and QGIS 3.10.0.

**Comment by SebastienPeillet** [comment]
Can't reproduce today in 3.10.2 (but with Ubuntu and not Windows), the fields are well appended without freeze.

@PedroVenancio , is it still happening ?

**Comment by PedroVenancio** [comment]
Hi @SebastienPeillet 

Tested now in Windows:

- QGIS version 3.4.15-Madeira, code revision e83d02e274
and the freeze is still true.

- QGIS version 3.10.2-A Coruña, code revision d971b12f63 
- QGIS version 3.11.0-Master, code revision 644a5525ec
and everything seems ok! I get no freeze!

So, I would say that this issue loose it's importance, because QGIS 3.4.x will become EOL very soon. 

What do you think?

**Comment by gioman** [comment]
>  issue loose it's importance

yes, because as far as I have understand there will not be another 3.4 point release (but I must admit I'm still a bit confused by the actual roadmap).

**Comment by PedroVenancio** [comment]
> yes, because as far as I have understand there will not be another 3.4 point release (but I must admit I'm still a bit confused by the actual roadmap).

It is also my understanding that there will not be another point release for 3.4 (after 3.4.15), and the LTR will be 3.10.3.

So, I will close this issue, since it will not be fixed to 3.4 series.
